apps/election-manager/src/lib/votecounting.ts

```diff
@@ -485,6 +385,171 @@ export const getContestTallyMeta = ({
+  const allVotes: VotesDict[] = []
+  const cvrMap = new Map<string, CastVoteRecord>()
+  let numberOfBallotsCounted = 0
+  castVoteRecords.forEach((CVR) => {
+    const vote = buildVoteFromCvr({ election, cvr: CVR })
+    cvrMap.set(CVR._ballotId, CVR)
+    allVotes.push(vote)
+    numberOfBallotsCounted += 1
+  })
+
+  const overallTally = tallyVotesByContest({ election, votes: allVotes })
+  const contestTallyMetadata = getContestTallyMeta({
+    election,
+    castVoteRecords,
+  })
+
+  return {
+    contestTallies: overallTally,
+    castVoteRecords: cvrMap,
+    numberOfBallotsCounted,
+    contestTallyMetadata,
+  }
+}
+
+export function computeFullElectionTally(
+  election: Election,
+  castVoteRecordLists: CastVoteRecordLists
+): OptionalFullElectionTally {
+  const castVoteRecords = castVoteRecordLists.flat(1)
+
+  const overallTally = getTallyForCastVoteRecords(election, castVoteRecords)
+
+  const cvrFilesByPrecinct: Dictionary<CastVoteRecord[]> = {}
+  const cvrFilesByScanner: Dictionary<CastVoteRecord[]> = {}
+  castVoteRecords.forEach((CVR) => {
+    let filesForPrecinct = cvrFilesByPrecinct[CVR._precinctId]
+    if (!filesForPrecinct) {
+      cvrFilesByPrecinct[CVR._precinctId] = filesForPrecinct = []
+    }
+    filesForPrecinct.push(CVR)
+
+    let filesForScanner = cvrFilesByScanner[CVR._scannerId]
+    if (!filesForScanner) {
+      cvrFilesByScanner[CVR._scannerId] = filesForScanner = []
+    }
+    filesForScanner.push(CVR)
+  })
+
+  const resultsByCategory = new Map<TallyCategory, Dictionary<Tally>>()
+  for (const category of Object.values(TallyCategory)) {
+    if (category === TallyCategory.Precinct) {
+      const precinctTallyResults: Dictionary<Tally> = {}
+      for (const precinctId in cvrFilesByPrecinct) {
+        const CVRs = cvrFilesByPrecinct[precinctId]!
+        const tally = getTallyForCastVoteRecords(election, CVRs)
+        precinctTallyResults[precinctId] = { ...tally, precinctId }
+      }
+      resultsByCategory.set(category, precinctTallyResults)
+    }
+
+    if (category === TallyCategory.Scanner) {
+      const scannerTallyResults: Dictionary<Tally> = {}
+      for (const scannerId in cvrFilesByScanner) {
+        const CVRs = cvrFilesByScanner[scannerId]!
+        const tally = getTallyForCastVoteRecords(election, CVRs)
+        scannerTallyResults[scannerId] = { ...tally, scannerId }
+      }
+      resultsByCategory.set(category, scannerTallyResults)
+    }
+  }
+
+  return {
+    overallTally,
+    resultsByCategory,
+  }
+}
+
+export function getEmptyTally(precinctId?: string, scannerId?: string): Tally {
+  return {
+    precinctId,
+    scannerId,
+    numberOfBallotsCounted: 0,
+    castVoteRecords: new Map<string, CastVoteRecord>(),
+    contestTallies: [],
+    contestTallyMetadata: {},
+  }
+}
+
+export function filterTalliesByParams(
+  fullElectionTally: FullElectionTally,
+  election: Election,
+  precinctId?: string,
+  scannerId?: string,
+  party?: Party
+): Tally {
```

Comment:
I think this would be easier to use if it took an options object.

```suggestion
export function filterTalliesByParams(
  fullElectionTally: FullElectionTally,
  election: Election,
  { precinctId, scannerId, party }: { precinctId?: string; scannerId?: string; party?: Party }
): Tally {
```